drivers/embedded/tensil/sample_buffer.c

```diff
@@ -137,55 +175,51 @@ error_t sample_buffer_print_analysis(
     memset(simd_flags_counts, 0, FLAGS_COUNTS_SIZE * sizeof(counter_t));
     memset(noop_flags_counts, 0, FLAGS_COUNTS_SIZE * sizeof(counter_t));
 
-    printf("Collected %zu samples\n", samples_count);
+    size_t valid_samples_count = 0;
+    uint32_t program_counter = 0;
+    uint32_t instruction_offset = 0;
+    const uint8_t *sample_ptr =
+        sample_buffer_find_valid_samples_ptr(sample_buffer);
+
+    while (sample_buffer_get_next_samples_ptr(
```

Comment:
Roughly how fast are we pulling samples now?

Reply:
Since we sample every 1000 cycles it is unnoticeable wrt memory latency and bandwidth. What affects total run-time in the driver is the need for it to be a multiple of sample block transfers. For example if sampling is 1000 cycles and block size is 1024, the run-time will be a multiple of 1,024,000 cycles. This can be reduced by reducing sampling rate (faster production of blocks) and by reducing block size or both.